pkg/clusteragent/externalmetrics/autoscaler_watcher_test.go

```diff
@@ -22,8 +22,14 @@ import (
 	kube_fake "k8s.io/client-go/kubernetes/fake"
 
 	"github.com/DataDog/datadog-agent/pkg/clusteragent/externalmetrics/model"
+	"github.com/DataDog/watermarkpodautoscaler/api/v1alpha1"
 )
 
+func init() {
+	autoscaler.AddToScheme(scheme)
+	v1alpha1.AddToScheme(scheme)
+}
+
```

Comment:
since the `scheme` is define in "package" not a test file. maybe it is better to move the init also in the "package"

Reply:
In fact, this `scheme` is defined in a test file of the same package, in [`pkg/clusteragent/externalmetrics/datadogmetric_controller_test.go` here](https://github.com/DataDog/datadog-agent/pull/7857/files#diff-f5f5499a84c7aeb5c77b7fcad8a533bc8832a16c29f3bdfb7293eea7445e941cR32).

Inside `pkg/clusteragent/externalmetrics`, there are 2 test files:
* `autoscaler_watcher_test.go` and
* `datadogmetric_controller_test.go`

As they belong to the same package, there is only one `scheme` variable (which happens to be declared in `datadogmetric_controller_test.go`.
* `autoscaler_watcher_test.go` does the `AddToScheme(…)` for the HPA and the WPA and
* `datadogmetric_controller_test.go` does the `AddToScheme(…)` for the DDM.

In the end, there’s only one `scheme` variable for the tests only but I’ve scattered the `AddToScheme(…)` to the test file which is needing a given type. Does that make sense ?